charts/lily/templates/cronjob-gapfill.yaml

```diff
@@ -0,0 +1,148 @@
+    {{- if .Values.labels }}
+    {{- toYaml .Values.labels | nindent 4 }}
+    {{- end }}
+spec:
+  # TODO: make cron schedule configurable
+  schedule: "0 0 * * *"
+  concurrencyPolicy: Forbid
+  jobTemplate:
+    metadata:
+      labels:
+        {{- include "sentinel-lily.releaseLabels" . | nindent 8 }}
+        {{- if .Values.labels }}
+        {{- toYaml .Values.labels | nindent 8 }}
+        {{- end }}
+    spec:
+      template:
+        metadata:
+          labels:
+            {{- include "sentinel-lily.releaseLabels" . | nindent 12 }}
+            {{- if .Values.labels }}
+            {{- toYaml .Values.labels | nindent 12 }}
+            {{- end }}
+        spec:
+          restartPolicy: OnFailure
+          {{- if .Values.imagePullSecrets }}
+          {{- with .Values.imagePullSecrets }}
+          imagePullSecrets:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          {{- end }}
+          # we're going to need root for a little while until
+          # we can package some helper binaries into the image
+          #securityContext:
+            #fsGroup: 562
+            #runAsNonRoot: true
+            #runAsUser: 562
+            #runAsGroup: 562
+          containers:
+          - name: gapfill
+            image: "{{ .Values.image.repo }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+            imagePullPolicy: {{ .Values.image.pullPolicy }}
+            command: ["/bin/bash", "-c"]
+            args:
+              - |
+                set -xo pipefail
+
+                function check(){
+                  if [ $1 -ne 0 ]; then
+                    exit $1
+                  fi
+                }
+
+                # TODO: include jq in lily image
+                apt-get update && apt-get install -y jq
+
+                lily wait-api --timeout=60s
+                check $?
+
+                ## TODO: check when last cron job was executed successfully
+                ## TODO: determine to/from window from last job exec
+                ## use default window of last 26 hours (overlap to
+                ## account for spin-up duration)
+
+                uid=`cat /var/lib/lily/uid`
+                to=$(((`date +%s` - {{ required "(root).daemon.genesisUnix expected" .Values.daemon.genesisUnix | int }}) / 30))
+                from=$(((`date -d "26 hours ago" +%s` - {{ required "(root).daemon.genesisUnix expected" .Values.daemon.genesisUnix | int }}) / 30))
+                echo "Starting gapfill jobs...from $from to $to"
+
+                {{ range .Values.daemon.gapfill.taskSets }}
+                {{- $jobPrint := include "sentinel-lily.fingerprintAllArgs" (printf "--tasks=%s" . | list ) }}
+                {{- $jobName := printf "%s/gapfind-${uid}/%s" (include "sentinel-lily.instanceName" $) $jobPrint }}
+                echo "lily gap find --from=$from --to=$to --tasks=\"{{ . }}\" --storage={{ $.Values.daemon.gapfill.storage }} --name \"{{ $jobName }}f=${from},t=${to}\""
```

Comment:
Move this to after the sync?

Reply:
This is just an echo. The next line blocks for sync and then the gap task is started.